File: 1535-Find-the-Winner-of-an-Array-Game/main.go
package main

func getWinner(arr []int, k int) int {

	if k >= len(arr) {
		max := arr[0]
		for _, v := range arr {
			if v > max {
				max = v
			}
		}
		return max
	}

	winner := arr[0]
	winCount := 0

	for i := 1; i < len(arr); i++ {
		if arr[i] > winner {
			winner = arr[i]
			winCount = 1
		} else {
			winCount += 1
		}

		if winCount == k {
			return winner
		}
	}

	return winner
}

File: 1535-Find-the-Winner-of-an-Array-Game/main.py
from typing import List

class Solution:
    def getWinner(self, arr: List[int], k: int) -> int:
        
        n = len(arr)

        if n <= k:
            return max(arr)
        
        count = 0
        winner = arr[0]
        for i in range(1, n):
            if arr[i]  > winner:
                winner = arr[i]
                count = 1
            else:
                count += 1
            
            if count == k:
                return winner
        
        return winner

